documentation/src/main/asciidoc/topics/proc_cdi_injecting_remote.adoc

```diff
@@ -0,0 +1,72 @@
+[id='cdi_inject_remote']
+= Injecting Remote Caches
+Set up CDI beans to inject remote caches.
+
+.Procedure
+
+. Create a cache qualifier annotation.
++
+[source,java]
+----
+@Remote("mygreetingcache") <1>
+@Qualifier
+@Target({ElementType.FIELD, ElementType.PARAMETER, ElementType.METHOD})
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+public @interface RemoteGreetingCache { <2>
+}
+----
++
+<1> names the cache to inject.
+<2> creates a `@RemoteGreetingCache` qualifier.
++
+. Add the `@RemoteGreetingCache` qualifier to your cache injection point.
++
+[source,java]
+----
+public class GreetingService {
+
+    @Inject @RemoteGreetingCache
+    private RemoteCache<String, String> cache;
+
+    public String greet(String user) {
+        String cachedValue = cache.get(user);
+        if (cachedValue == null) {
+            cachedValue = "Hello " + user;
+            cache.put(user, cachedValue);
+        }
+        return cachedValue;
+    }
+}
+----
+
+[TIP]
+====
+You can also inject remote caches without using qualifiers.
+
+[source,java]
+----
+...
+   @Inject
+   @Remote("greetingCache")
+   private RemoteCache<String, String> cache;
+----
+====
+
+.Override the default remote cache manager
+
+[source,java]
+----
+...
+import javax.enterprise.context.ApplicationScoped;
+
+public class Config {
+
+    @RemoteGreetingCache
+    @Produces
+    @ApplicationScoped
+    public RemoteCacheManager defaultRemoteCacheManager() {
+        return new RemoteCacheManager(localhost, 1544);
```

Comment:
@danberindei should this be `RemoteCacheManager()` without the (string, int)?

as you say there is no configuration with remote caches. `RemoteCacheManager()` looks up the list of servers in a file hotrod-client.properties in the classpath

Reply:
Yes, that's I wanted to say initially, but a more appropriate example would be
```
ConfigurationBuilder builder = new ConfigurationBuilder();
builder.addServer().host("localhost").port(11222);
return new RemoteCacheManager(builder.build());
```